pkg/vsock/handshake_windows.go

```diff
@@ -48,76 +49,73 @@ func GetVMGUID(seed string, handshakePort uint32, timeout <-chan time.Time) (hvs
 
 	key.Close()
 
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
We should pass the context down from `main` rather than using `context.Background`.

Reply:
This is intentional, the `cancel` here is called once the VM GUID is found to cancel out all the goroutines that are performing the handshake, if we pass in the context from main this will cancel all the core processes in main, eg. `run()`, `httpServe`, etc